src/domain/valueObject/username.go

```diff
@@ -3,33 +3,37 @@ package valueObject
 import (
 	"errors"
 	"regexp"
+
+	voHelper "github.com/speedianet/os/src/domain/valueObject/helper"
 )
 
 const usernameRegex string = `^[a-z_]([a-z0-9_-]{0,31}|[a-z0-9_-]{0,30}\$)$`
 
 type Username string
 
-func NewUsername(value string) (Username, error) {
-	user := Username(value)
-	if !user.isValid() {
+func NewUsername(value interface{}) (username Username, err error) {
+	stringValue, err := voHelper.InterfaceToString(value)
+	if err != nil {
+		return username, errors.New("UsernameValueMustBeString")
+	}
+
+	re := regexp.MustCompile(usernameRegex)
+	isValid := re.MatchString(stringValue)
+	if !isValid {
 		return "", errors.New("InvalidUsername")
 	}
-	return user, nil
+	return Username(stringValue), nil
 }
 
-func NewUsernamePanic(value string) Username {
-	user := Username(value)
-	if !user.isValid() {
-		panic("InvalidUsername")
+// TODO: remove this constructor when no longer used.
```

Comment:
Same as [this comment](https://github.com/speedianet/os/pull/142/files#r1683023375).

Reply:
Done.